module2/lessons/js_intro_to_dom_manipulation.md

```diff
@@ -0,0 +1,393 @@
+
+This power emerges when we start **listening for user events**. This is the crux of front-end engineering. **We present a user interface and then as the user interacts with the UI, we change and update what they see.**
+
+<section id="try-it" class="call-to-action">
+### Try It Together
+
+Let's add an eventListener to our Report Card codepen. We still want to change the student name and grade, but we want to do so only when the user takes an action.  Let's say we want to change that data if a user clicks on the "Report Card" title
+  
+- We'll use `document.querySelector()` to grab the "Report Card" `<h1>` and save it to a variable
+- Now we can add an eventListener to that `<h1>` title.
+  - We want the eventListener to listen for a "click" event that happens on that title element. 
+  - When the "click" happens on the title element, we want to invoke a function that does all the things we want to do. Let's call that function `editReportCard`.
+- You've already written the code to change the student name and grade.  Now we can move this code into the `editReportCard` function that is being invoked by the eventListener.
+
+Consider:  
+- Why don't we use `()` to invoke the function within the eventListener?  What happens if we do?
+</section>
+
+<section class="call-to-action">
+### Your Turn
+
+In the codepen from the Warm Up, follow the bullets below to add in an Event Listener. We still want to change the title, fact, and image, but we want to do so only when the user clicks a button.  
+  
+- Add a "Change Animal" button to your html above the <h1> element. Give it an id of  'change-animal-btn'.
+- In your js file, use `document.querySelector()` to grab that button by its id and save it to a variable
+- Now add an eventListener to the button.
+  - We want the eventListener to listen for a "click" event that happens on that button. 
+  - When the "click" happens on the button, invoke a function that does all the things we want to do. Call that function `changePageContent`.
+- You've already written the code to change the title, fact and image.  Move this code into the `changePageContent` function that is being invoked by the eventListener.
+</section>
+
+### Explore
+
+There are many types of events beyond just "click".   Explore this list of [Mouse Events](https://developer.mozilla.org/en-US/docs/Web/API/Element#mouse_events) on MDN. Scroll beyond just mouse events to explore keyboard events, focus events, etc.  
+
+Think of a webpage you often visit. Maybe it's a social media page, maybe it's your bank's website, maybe it's a news site. Take a moment and consider all the event listeners that might be on that page! What type of events might they be listening for? What functionality occurs when those events happen? 
+
+## More Programmatic DOM Manipulation
+
+### `innerHTML`
```

Comment:
So glad to see that we're calling out the difference between innerText and innerHTML - it's a common point of confusion for students